cmd/hope/node/utils_test.go

```diff
@@ -0,0 +1,79 @@
+package node
+
+import (
+	"testing"
+)
+
+import (
+	"github.com/spf13/viper"
+	"github.com/stretchr/testify/assert"
+)
+
+import (
+	"github.com/Eagerod/hope/pkg/hope"
+)
+
+// Basically a smoke test, don't want to define a ton of yaml blocks to test
+//   this extensively quite yet.
+func TestGetNodes(t *testing.T) {
+	// Assume config file in the project root.
+	// Probably bad practice, but better test than having nothing at all.
+	viper.AddConfigPath("../../../")
+	viper.SetConfigName("hope")
+	viper.AutomaticEnv()
+
+	err := viper.ReadInConfig()
+	assert.Nil(t, err)
+
+	nodes, err := getNodes()
+	assert.Nil(t, err)
+	assert.Equal(t, 2, len(*nodes))
+
+	expected := []hope.Node{
+		hope.Node{
+			Name: "home-master-01",
+			Role: "master",
+			Host: "192.168.1.31",
+			User: "root",
+		},
+		hope.Node{
+			Name: "home-node-01",
+			Role: "node",
+			Host: "192.168.1.30",
+			User: "root",
+		},
+	}
+
+	assert.Equal(t, *nodes, expected)
+}
+
+func TestGetNode(t *testing.T) {
+	// Assume config file in the project root.
+	// Probably bad practice, but better test than having nothing at all.
+	viper.AddConfigPath("../../../")
```

Comment:
Might be worth making helper functions in test files to do these ~5 statements that pop up in every test.

Reply:
Did this for a different reason, but it happened.